core-bundle/src/Resources/contao/pages/PageRegular.php

```diff
@@ -808,6 +811,48 @@ protected function createFooterScripts($objLayout)
 			}
 		}
 
+		// Add search index meta data
+		if ($objPage !== null)
+		{
+			$noSearch = (bool) $objPage->noSearch;
+
+			// Backwards compatibility: Do not use $GLOBALS['TL_NOINDEX_KEYS'] anymore
+			// but make sure your page type delivers the correct meta robots tag
+			foreach (array_keys($_GET) as $key)
+			{
+				foreach ($GLOBALS['TL_NOINDEX_KEYS'] as $noIndexKey)
+				{
+					if (preg_match('/^' . $noIndexKey . '$/', $key))
+					{
+						$noSearch = true;
+						break 2;
+					}
+				}
+			}
+
+			$meta = array
+			(
+				'@context' => 'https://contao.org/',
+				'@type' => 'PageMetaData',
+				'pageId' => (int) $objPage->id,
+				'language' => $objPage->language,
+				'title' => $objPage->pageTitle ?: $objPage->title,
+				'noSearch' => $noSearch,
+				'protected' => (bool) $objPage->protected,
+				'groups' => array_map('intval', array_filter((array) $objPage->groups)),
+				'fePreview' => System::getContainer()->get('contao.security.token_checker')->isPreviewMode()
+			);
+
+			$token = System::getContainer()->get('security.token_storage')->getToken();
```

Comment:
I think we should use the `security.helper` service here instead, shouldn't we? @aschempp /cc

Reply:
Doesn't really make things much shorter but done in 7f8a3d704110615cd31b5785f3df03245106391b.